src/Umbraco.PublishedCache.NuCache/ContentCache.cs

```diff
@@ -14,442 +11,441 @@
 using Umbraco.Cms.Infrastructure.PublishedCache.Navigable;
 using Umbraco.Extensions;
 
-namespace Umbraco.Cms.Infrastructure.PublishedCache
+namespace Umbraco.Cms.Infrastructure.PublishedCache;
+
+public class ContentCache : PublishedCacheBase, IPublishedContentCache, INavigableData, IDisposable
 {
-    public class ContentCache : PublishedCacheBase, IPublishedContentCache, INavigableData, IDisposable
+    private readonly IDomainCache _domainCache;
+    private readonly IAppCache? _elementsCache;
+    private readonly GlobalSettings _globalSettings;
+    private readonly ContentStore.Snapshot _snapshot;
+    private readonly IAppCache _snapshotCache;
+    private readonly IVariationContextAccessor _variationContextAccessor;
+
+    #region IDisposable
+
+    public void Dispose() => _snapshot.Dispose();
+
+    #endregion
+
+    #region Constructor
+
+    // TODO: figure this out
+    // after the current snapshot has been resync-ed
+    // it's too late for UmbracoContext which has captured previewDefault and stuff into these ctor vars
+    // but, no, UmbracoContext returns snapshot.Content which comes from elements SO a resync should create a new cache
+    public ContentCache(
+        bool previewDefault,
+        ContentStore.Snapshot snapshot,
+        IAppCache snapshotCache,
+        IAppCache? elementsCache,
+        IDomainCache domainCache,
+        IOptions<GlobalSettings> globalSettings,
+        IVariationContextAccessor variationContextAccessor)
+        : base(previewDefault)
     {
-        private readonly IDomainCache _domainCache;
-        private readonly IAppCache? _elementsCache;
-        private readonly GlobalSettings _globalSettings;
-        private readonly ContentStore.Snapshot _snapshot;
-        private readonly IAppCache _snapshotCache;
-        private readonly IVariationContextAccessor _variationContextAccessor;
+        _snapshot = snapshot;
+        _snapshotCache = snapshotCache;
+        _elementsCache = elementsCache;
+        _domainCache = domainCache ?? throw new ArgumentNullException(nameof(domainCache));
+        _globalSettings = globalSettings.Value;
+        _variationContextAccessor = variationContextAccessor;
+    }
 
-        #region IDisposable
+    private bool HideTopLevelNodeFromPath => _globalSettings.HideTopLevelNodeFromPath;
```

Comment:
This is in the constructor region but it isn't a constuctor, should probably just remove all regions from the file